scripts/Units/armfido.bos

```diff
@@ -352,21 +327,33 @@ Create()
 	move pelvis to y-axis 4000 speed <500>; 
 	restore_delay = 30;
 	animspeed = 4;
-	Static_Var_2 = 0;
-	gun_2 = barrel;
-	statechg_DesiredState = TRUE;
-	statechg_StateChanging = FALSE;
+	Static_Var_2 = 1;
 	moveSpeed = get MAX_SPEED;
+	// set last heading values to large negative, ensure first aim waits for the turn
+	last_primary_heading = -1000000;
+	last_primary_pitch = -1000000;
 }
 
 Activate()
 {
-	start-script RequestState(0);
+	// signal kills any current aim animations
```

Comment:
Comments explaining `signal` and `set-signal-mask` aren't generally needed, as this is base BOS functionality

Reply:
It serves to explain why signal is used here, so it is not removed in the future by accident. When armfido.bos was transitioned for unit_continuous_aim.lua compatibility, signals were stripped out leading to animation errors, two or more turns trying to complete.